### PR TITLE
extended DEVELOPERS.md with tunnel config

### PR DESCRIPTION
**Pull request by ideesnoires**
just added an example block for wireguard tunnels

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -114,6 +114,17 @@ networks:
     assignments:              # assign static addresses to devices
       magda-core: 1
 
+# not part of magda conf, only to illustrate tunnel usage
```
Comment (Noki) [inline DEVELOPER.md:117]: please put the example in a seperate code block

File: DEVELOPER.md [hunk 2]
```diff
@@ -114,6 +114,17 @@ networks:
...
+  - vid: 13
+    untagged: true            # don't tag this vlan - useful if the corerouter is plugged into a normal home router
+    name: uplink
+    role: ext
+    tunnel_wan_ip: 192.168.1.2/24   # put here the address of the corerouter inside the uplink network
+    tunnel_wan_gw: 192.168.1.1      # gateway of the uplink network
+    tunnel_connections: 2           # default value, number of different tunnels to create
+    tunnel_timeout: 600             # timeout in seconds after this the tunnel is destroyed and attempted to be rebuild
+    tunnel_mesh_prefix_ipv4: 10.31.142.120/29   # ip address space for both sides of the tunnel
```
Comment (Noki) [inline DEVELOPER.md:126]: This should be explained in more detail. What is the size requirement for this ip-block? Is it tunnel_connections * 2? In addition it should also clearly state that this is a subnet derived from the main ip-block for this node.

File: DEVELOPER.md [hunk 3]
```diff
@@ -114,6 +114,17 @@ networks:
     assignments:              # assign static addresses to devices
       magda-core: 1
 
+# not part of magda conf, only to illustrate tunnel usage
+  - vid: 13
+    untagged: true            # don't tag this vlan - useful if the corerouter is plugged into a normal home router
```
Comment (Noki) [inline DEVELOPER.md:119]: can this also be a tagged vlan? can I limit this vlan to a single port?

File: DEVELOPER.md [hunk 4]
```diff
@@ -114,6 +114,17 @@ networks:
...
+    name: uplink
+    role: ext
+    tunnel_wan_ip: 192.168.1.2/24   # put here the address of the corerouter inside the uplink network
```
Comment (Noki) [inline DEVELOPER.md:122]: this is a network, not only an ip. I suggest to change this to "put the addess of the core router inside the uplink network here together with the netmask of the uplink network.

**Review by Noki — CHANGES_REQUESTED** [review]

File: DEVELOPER.md (hunk 1 above)
Comment (ideesnoires) [reply DEVELOPER.md:117]: i put there because it's part of the networks.yml so how can i put it into an extra block?

File: DEVELOPER.md (hunk 2 above)
Comment (ideesnoires) [reply DEVELOPER.md]: was technically wrong, now fixed. this is only the ip of the corerouter end of the tunnel

File: DEVELOPER.md (hunk 3 above)
Comment (ideesnoires) [reply DEVELOPER.md]: explained a bit more. can also be tagged. afaik not possible to assign to a specific port.

File: DEVELOPER.md (hunk 1 above)
Comment (ideesnoires) [reply DEVELOPER.md:117]: i mean should i put it into an extra section like: uplink via tunnel -> your networks.yml is gonna look like this?

File: DEVELOPER.md [hunk 5]
```diff
@@ -130,6 +130,23 @@ The VLAN ID (vid) usually follow this numbering convention.
...
+```yml
+  - vid: 13
+    untagged: true            # option to don't tag this vlan - useful if the corerouter is plugged into a normal home router
+    name: uplink
+    role: ext
+    tunnel_wan_ip: 192.168.1.2/24   # put here the address and subnet of the corerouter inside the uplink network
+    tunnel_wan_gw: 192.168.1.1      # gateway of the uplink network
+    tunnel_connections: 2           # default value, number of different tunnels to create
+    tunnel_timeout: 600             # timeout in seconds after this the tunnel is destroyed and attempted to be rebuild
+    tunnel_mesh_prefix_ipv4: 10.31.142.120/32   # ip address for this side of the tunnel
```
Comment (PolynomialDivision) [inline DEVELOPER.md:147]: This config will not work. You have a "pool" of 1 address because of /32 but you want 2 tunnel_connections.

**Review by Noki — APPROVED** [review]

**Comment by PolynomialDivision** [comment]
Please squash the commits together

**Pull request merged**